src/devices/vdml_usd.cpp

```diff
@@ -21,5 +21,80 @@ std::int32_t is_installed(void) {
 	return usd_is_installed();
 }
 
+std::int32_t list_files_raw(const char* path, char* buffer, int32_t len) {
+	return usd_list_files_raw(path, buffer, len);
+}
+
+std::vector<std::string> list_files(const char* path, char* buffer, int32_t len) {
+	std::vector<std::string> files = {};
+	// Call the C function
+	int32_t success = usd_list_files_raw(path, buffer, len);
+	// Check if call successful, if error return vector containing error state
+	if (success == PROS_ERR) {
+		// Check errno to see which error state occurred
+		// push back error state to files vector as std::string
+		switch (errno) {
+			case EIO:
+				files.push_back("EIO");
+				break;
+			case EINVAL:
+				files.push_back("EINVAL");
+				break;
+			case EBUSY:
+				files.push_back("EBUSY");
+				break;
+			case ENOENT:
+				files.push_back("ENOENT");
+				break;
+			case EACCES:
+				files.push_back("EACCES");
+				break;
+			case EEXIST:
+				files.push_back("EEXIST");
+				break;
+			case EROFS:
+				files.push_back("EROFS");
+				break;
+			case ENXIO:
+				files.push_back("ENXIO");
+				break;
+			case ENOBUFS:
+				files.push_back("ENOBUFS");
+				break;
+			case ENFILE:
+				files.push_back("ENFILE");
+				break;
+			default:
+				// If none of the above, will be FILE IO error
+				files.push_back("FILE I/O ERROR");
+				break;
+		}
+		return files;
+	}
+	// Parse buffer given call successful, split by '/n'
+	// Store char * buffer as std::string in str
+	// Store delimiter in std::string, '\n'
+	std::string str(buffer);
+	std::string delimiter = "\n";
```

Comment:
```cpp
char delimeter = '\n';
```

You definitely don't need a whole `std::string` for just `\n`, which is only a single char.

Reply:
Implemented this change in commit 3d41ca62de29da852509b02fcd8ce85c94a92e60